### ISSUE TITLE
Inline @var syntax

### ISSUE THREAD

**Issue by kokx** [mkusher/padawan.php#73]
Implement inline `@var` syntax.

Sometimes, you know for a fact that some function returns a certain type. However, type hinting won't work because the function is too generic for that. For example, with the ZF2 ServiceManager:

```php
<?php
use My\Job;

$job = $sm->get(Job::class);
/* @var $job Job */

$job->...
```

In this case, it would be great to have autocompletion for `$job` as well. However, padawan.php just marks it as a generic `object`.

This syntax has been implemented by many IDE's, for example [NetBeans](https://blogs.oracle.com/netbeansphp/entry/defining_a_variable_type_in).

**Comment by mkusher** [comment]
@kokx now `@var` works only before assignment, like this
```php
<?php
use My\Job;

/* @var $job Job */
$job = $sm->get(Job::class);

$job->... // type of $job is Job here
```

**Comment by kokx** [comment]
Thanks for the comment @mkusher. Unfortunately however, I cannot reproduce that. Even after a re-index and restart of padawan.php it doesn't work. It still tells me that `$job` is an `object`.

**Comment by pbogut** [comment]
Shouldn't it be `/** @var $job Job */` instead of `/* @var $job Job */` ?
Note two asterisks.

**Comment by mkusher** [comment]
oh, @pbogut is right :)

**Comment by kokx** [comment]
@pbogut That gives a little different result. It now assigns the current namespace (without class) to the variable. See this screenshot:

![var-inline-mistake](https://cloud.githubusercontent.com/assets/138556/21391939/fc0cab54-c78d-11e6-9892-3f27ee26be2c.png)
(the current namespace here is `Debtor\Service`).

Using `/** @var Job $job */` does work correctly. So thanks for the hint!

Still, I think it is a good idea to implement the same syntax as NetBeans and PhpStorm, for teams in which different editors are used (such as my team).

**Comment by mkusher** [comment]
@kokx now padawan partially supports doc-comments, which are using double asterisk(and phpstorm understands this syntax too).

**Comment by kokx** [comment]
@mkusher Yeah, I completely get that. I'm just saying that it would be a nice feature to have (imo not top priority). Since a lot of code bases already use that syntax.

**Comment by halftan** [comment]
will implement in #80, @kokx @pbogut please checkout my development branch to test the feature.
The inline type hinting formats are:
```php
// Eclipse's approach of type hint
/** @var $inst Class */
$inst = new $className();

// phpdoc-like approach
for ($items as $item) {
    /** @var Class $item */
}

// also works in closure
array_map($array, function($a, $b) {
    /** @var Class $a */
    /** @var Class $b */
    // or maybe in block ( not available now)
    /**
     * @var Class $a
     * @var Class $b
     **/
});
```
